libtransact/src/state/btree.rs

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 Bitwise IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+use crate::database::error::DatabaseError;
+use crate::database::lmdb::{Database, DatabaseReader, DatabaseWriter, LmdbDatabaseReaderCursor};
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+#[derive(Clone)]
+pub struct BTreeBackEnd {
```

Comment:
Suggestions on how to name BTreeBackEnd? This is just a wrapper to the BTreeDatabase so the whole struct is contained in a single mutex. `BTreeBackEnd` name was more of a placeholder

Reply:
This should be called `BTreeDatabase`, and the internal struct should have a name reflecting (Like BTreeDbInternal, or something).  The internal struct should also be module-private